### PR TITLE
All the sandbox containers ready to move to containers. 

### PR DESCRIPTION
**Pull request by ypriverol**
@prvst or @osallou can you review this PR. When you accept it I will delete the sandbox repository and only have the **containers** repository for the future.

### CODE REVIEW

File: DiffExpIR/0.0.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# Metadata
+LABEL base.image="bamtools:latest"
+LABEL version="3"
+LABEL software="DiffExpIR"
+LABEL software.version="0.0.1"
+LABEL description="Differentially expressed intron retention"
+LABEL website="https://github.com/r78v10a07/DiffExpIR"
+LABEL documentation="https://github.com/r78v10a07/DiffExpIR"
+LABEL license="http://www.gnu.org/licenses/"
```
Comment (osallou) [inline DiffExpIR/0.0.1/Dockerfile]: license is not explicit, should specify which license

File: Rstudio-Proteomics/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# Provides:         Rstudio
+# Base Image:       bioconductor/release_core
+# Build Cmd:        docker build --rm -t biodckr/rstudio .
+# Pull Cmd:         docker pull biodckr/rstudio
+# Run Cmd:          docker run --rm  -p 8787:8787 biodckr/rstudio
+# Extra:            The Rstudio will be available via browser at:
+# Extra:            http://localhost:8787
+# Extra:            user: rstudio
+# Extra:            pass: rstudio
+#################################################################
```
Comment (osallou) [inline Rstudio-Proteomics/Dockerfile]: should use Docker LABEL to define most of those metadata and should specify license

File: TPMCalculator/0.0.1/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# Metadata
+LABEL base.image="bamtools:latest"
+LABEL version="3"
+LABEL software="TPMCalculator"
+LABEL software.version="0.0.1"
+LABEL description="This program calculates the TPM (Transcript per Millions) values for the exons and introns from NGS RNA-Seq aligned reads (BAM files)"
+LABEL website="https://github.com/r78v10a07/TPMCalculator"
+LABEL documentation="https://github.com/r78v10a07/TPMCalculator"
+LABEL license="http://www.gnu.org/licenses/"
```
Comment (osallou) [inline TPMCalculator/0.0.1/Dockerfile:12]: again, should specify which license

File: bior_annotate/2.1.1/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+# Extra:            Data/
+# Extra:              references/
+# Extra:                hg19.fa.gz
+# Extra:                hg19.fa.gz.fai
+# Extra:            catalogs/
+# Extra:              2015_05_18/
+# Extra:                noTCGA_ExAc.datasource.properties
+# Extra:                noTCGA_ExAc.columns.tsv
+# Extra:                noTCGA_ExAc.tsv.bgz.tbi
+# Extra:                noTCGA_ExAc.tsv.bgz
```
Comment (osallou) [inline bior_annotate/2.1.1/Dockerfile:36]: should use LABEL for most of those metadata, missing license info

File: discovar/52488/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+# Software Version: 52488
+# Description:      DISCOVAR is a new variant caller
+# Website:          http://www.broadinstitute.org/software/discovar/blog/?page_id=19
+# Tags:             Genomics
+# Provides:         discovar 52488
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/discovar 52488/.
+# Pull Cmd:         docker pull biodckrdev/discovar
+# Run Cmd:          docker run --rm -it biodckrdev/discovar <options> <files>
+#################################################################
```
Comment (osallou) [inline discovar/52488/Dockerfile]: needs to be in LABEL and specify license

File: discovar/52488/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline discovar/52488/Dockerfile]: should not use biocontainers image?

File: discovardenovo/52488/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+# Description:      Genome variant discovery through assembly
+# Website:          http://www.broadinstitute.org/software/discovar/blog/?page_id=19
+# Tags:             Genomics
+# Provides:         discovardenovo 52488
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/discovardenovo 52488/.
+# Pull Cmd:         docker pull biodckrdev/discovardenovo
+# Run Cmd:          docker run --rm -it biodckrdev/discovardenovo <options> <files>
+#################################################################
+
```
Comment (osallou) [inline discovardenovo/52488/Dockerfile]: must use LABEL and add license info

File: discovardenovo/52488/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline discovardenovo/52488/Dockerfile]: can't use biocontainers base image?

File: entrez-direct/7.50.20171103/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+# Metadata
+LABEL base.image="biocontainers:latest"
+LABEL version="3"
+LABEL software="entrez-direct"
+LABEL software.version="7.50.20171103"
+LABEL description="Entrez Direct (EDirect) is an advanced method for accessing the NCBI's set of interconnected Entrez databases"
+LABEL website="https://github.com/ncbi/magicblast"
+LABEL documentation="https://ncbi.github.io/magicblast/"
+LABEL License_file = /home/biodocker/bin/LICENSE
+LABEL tags="BLAST"
```
Comment (osallou) [inline entrez-direct/7.50.20171103/Dockerfile]: would be nice to ad license in addition to license_file to know what is the license type

File: frc_align/20150723/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+# Software Version: 20150723
+# Description:      Computes FRC from SAM/BAM file and not from afg files
+# Website:          https://github.com/jts/frc_courve
+# Tags:             Genomics
+# Provides:         frc_align 20150723
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/frc_align 20150723/.
+# Pull Cmd:         docker pull biodckrdev/frc_align
+# Run Cmd:          docker run --rm -it biodckrdev/frc_align <options> <files>
+#################################################################
```
Comment (osallou) [inline frc_align/20150723/Dockerfile]: must use LABEL and add license info

File: frc_align/20150723/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline frc_align/20150723/Dockerfile]: can't use biocontainers image?

File: gapcloser/1.12-r6/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+# Tags:             Genomics
+# Provides:         gapcloser 1.12-r6
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/gapcloser v1.12-r6/.
+# Pull Cmd:         docker pull biodckrdev/gapcloser
+# Run Cmd:          docker run --rm -it biodckrdev/gapcloser <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline gapcloser/1.12-r6/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: gatk/3.4/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+# Software Version: 3.4
+# Description:      The Genome Analysis Toolkit or GATK is a software package developed at the Broad Institute to analyze high-throughput sequencing data.
+# Website:          https://www.broadinstitute.org/gatk/|https://github.com/broadgsa/gatk-protected/
+# Tags:             Genomics
+# Provides:         gatk 3.4|samtools 1.2|bwakit 0.7.12|picard tools 1.139
+# Base Image:       java:7
+# Build Cmd:        docker build --rm -t biodckrdev/gatk 3.4/.
+# Pull Cmd:         docker pull biodckrdev/gatk
+# Run Cmd:          docker run --rm -it biodckrdev/gatk <options> <files>
+#################################################################
```
Comment (osallou) [inline gatk/3.4/Dockerfile]: same remarks, should use LABEL for metadata (+ license info)

File: gcta/1.24.7/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+# Tags:             Genomics
+# Provides:         gcta 1.24.7
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/gcta 1.24.7/.
+# Pull Cmd:         docker pull biodckrdev/gcta
+# Run Cmd:          docker run --rm -it biodckrdev/gcta <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline gcta/1.24.7/Dockerfile:18]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: htslib/1.2.1/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+# Provides:         htslib 1.2.1
+# Tags:             Genomics
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/htslib 1.2.1/.
+# Pull Cmd:         docker pull biodckrdev/htslib
+# Run Cmd:          docker run --rm -it biodckrdev/htslib <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline htslib/1.2.1/Dockerfile:18]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: jbrowse/1.11.6/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+# Tags:             Genomics
+# Provides:         jbrowse 1.11.6
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/jbrowse 1.11.6/.
+# Pull Cmd:         docker pull biodckrdev/jbrowse
+# Run Cmd:          docker run --rm -p 8000:8000 biodckrdev/jbrowse
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline jbrowse/1.11.6/Dockerfile:18]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: jellyfish/2.2.3/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+# Tags:             Genomics
+# Provides:         jellyfish 2.2.3
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/jellyfish 2.2.3/.
+# Pull Cmd:         docker pull biodckrdev/jellyfish
+# Run Cmd:          docker run --rm -it biodckrdev/jellyfish <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline jellyfish/2.2.3/Dockerfile:18]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: magicblast/1.3.0/Dockerfile [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+
+# Metadata
+LABEL base.image="biocontainers:latest"
+LABEL version="3"
+LABEL software="magicblast"
+LABEL software.version="1.3.0"
+LABEL description="Magic-BLAST is a new tool for mapping large sets of next-generation RNA or DNA sequencing runs against a whole genome or transcriptome"
+LABEL website="https://github.com/ncbi/magicblast"
+LABEL documentation="https://ncbi.github.io/magicblast/"
+LABEL License_file = /home/biodocker/bin/LICENSE
```
Comment (osallou) [inline magicblast/1.3.0/Dockerfile:12]: should add License tag to specify which license type it is

File: raremetalworker/4.13.7/Dockerfile [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+# Tags:             Genomics
+# Provides:         raremetalworker 4.13.7
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/raremetalworker 4.13.7/.
+# Pull Cmd:         docker pull biodckrdev/raremetalworker
+# Run Cmd:          docker run --rm -it biodckrdev/raremetalworker <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline raremetalworker/4.13.7/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: sga/0.10.13/Dockerfile [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+# Tags:             Genomics
+# Provides:         sga 0.10.13|bamtools 2.4.0|sparsehash 2.0.2|jemalloc 4.0.2|pysam 0.8.3|ruffus 2.6.3
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/sga 0.10.13/.
+# Pull Cmd:         docker pull biodckrdev/sga
+# Run Cmd:          docker run --rm -it biodckrdev/sga <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline sga/0.10.13/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: snpeff/4.1k/Dockerfile [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+# Tags:             Genomics
+# Provides:         snpeff 4.1k
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/snpeff 4.1k/.
+# Pull Cmd:         docker pull biodckrdev/snpeff
+# Run Cmd:          docker run --rm -it biodckrdev/snpeff <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline snpeff/4.1k/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: soapec/2.03/Dockerfile [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+# Tags:             Genomics
+# Provides:         soapec 2.03
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/soapec 2.03/.
+# Pull Cmd:         docker pull biodckrdev/soapec
+# Run Cmd:          docker run --rm -it biodckrdev/soapec <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline soapec/2.03/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: spades/3.6.0/Dockerfile [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+# Tags:             Genomics
+# Provides:         spades 3.6.0
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/spades 3.6.0/.
+# Pull Cmd:         docker pull biodckrdev/spades
+# Run Cmd:          docker run --rm -it biodckrdev/spades <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline spades/3.6.0/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: stacks/1.41/Dockerfile [hunk 24]
```diff
@@ -0,0 +1,55 @@
...
+# Tags:             Genomics
+# Provides:         Stacks 1.41
+# Base Image:       biodckr/biodocker:latest
+# Build Cmd:        docker build --rm -t biodckr/stacks 1.41/.
+# Pull Cmd:         docker pull biodckr/stacks
+# Run Cmd:          docker run --rm biodckr/stacks <tool> <options>
+#################################################################
+
+# Set the base image to biodckr/biodocker
+FROM biodckr/biodocker:latest
```
Comment (osallou) [inline stacks/1.41/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: tophat/2.1.1/Dockerfile [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+# Tags:             Genomics
+# Provides:         tophat 2.2.1
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/tophat 2.1.1/.
+# Pull Cmd:         docker pull biodckrdev/tophat
+# Run Cmd:          docker run --rm -it biodckrdev/tophat <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline tophat/2.1.1/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

File: vcf-miner/4.0.1/Dockerfile [hunk 26]
```diff
@@ -0,0 +1,47 @@
+FROM centos:6
```
Comment (osallou) [inline vcf-miner/4.0.1/Dockerfile]: No LABEL at all, needs base and license info

File: velvet/1.2.10/Dockerfile [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+# Tags:             Genomics
+# Provides:         velvet 1.2.10
+# Base Image:       biodckr/biodocker
+# Build Cmd:        docker build --rm -t biodckrdev/velvet 1.2.10/.
+# Pull Cmd:         docker pull biodckrdev/velvet
+# Run Cmd:          docker run --rm -it biodckrdev/velvet <options> <files>
+#################################################################
+
+# Set the base image to Ubuntu
+FROM biodckr/biodocker
```
Comment (osallou) [inline velvet/1.2.10/Dockerfile]: same remarks, should use biocontainers base image and use LABEL for metadata (+ license info)

**Review by osallou — CHANGES_REQUESTED** [review]
most remarks about LABEL missing and missing license.

bior_annotate includes lots of code, should it really be merged in this repo?

**Comment by ypriverol** [comment]
@osallou I will review all the containers you have mentioned and update my PR.

**Comment by osallou** [comment]
regading licenses, we proposed change:

License = type of license
License_file = link to license file in container (or remote if not required to be in container)

**Comment by ypriverol** [comment]
I agree with your but I would like to do this in steps because this is a major change in all the containers.

**Comment by ypriverol** [comment]
@osallou @prvst please review the current PR. It should be ready with major changes in most of the containers.

**Review by prvst — APPROVED** [review]

**Pull request merged**